src/main/java/com/jvm_bloggers/admin_panel/mailing/MailingAddressUniquenessValidator.java

```diff
@@ -0,0 +1,46 @@
+package com.jvm_bloggers.admin_panel.mailing;
+
+import com.jvm_bloggers.core.mailing.domain.MailingAddressRepository;
+import org.apache.wicket.markup.html.form.Form;
+import org.apache.wicket.markup.html.form.FormComponent;
+import org.apache.wicket.markup.html.form.validation.AbstractFormValidator;
+import org.apache.wicket.spring.injection.annot.SpringBean;
+import org.apache.wicket.util.lang.Args;
+
+import java.util.Optional;
+
+public class MailingAddressUniquenessValidator extends AbstractFormValidator {
```

Comment:
I would be much simpler if you could try to use IValidator add add it directly to textfield, sth like this:

    public class UsernameValidator implements IValidator<String> {
	List<String> existingUsernames = Arrays.asList("bigJack", "anonymous", "mrSmith");
	public void validate(IValidatable<String> validatable) {
		String chosenUserName = validatable.getValue();

		if(existingUsernames.contains(chosenUserName)){
			ValidationError error = new ValidationError(this);
			Random random = new Random();

			error.setVariable("suggestedUserName", 
					validatable.getValue() + random.nextInt());
			validatable.error(error);
		}
	}	
}

taken from https://ci.apache.org/projects/wicket/guide/7.x/guide/forms2.html#forms2_2, section "Creating custom validators". 

This way you won't be forced to pass components to validator and cast them to `FormComponent`